### PR TITLE
Replace `gulp serve` with `npm run serve`

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR partly fell out of the https://github.com/alphagov/govuk-frontend/pull/2861 work, and makes a few changes:

1. Replaces the `gulp serve` task with `npm run serve`
2. Adds a **nodemon.json** config file
4. Prevent Nodemon restarts for test files

For example, we don't need the review application Node.js process to restart when `checkboxes.test.js` is saved

Related to https://github.com/alphagov/govuk-frontend/issues/2717

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -52,24 +51,15 @@ gulp.task('compile', gulp.series(
...
-
 /**
  * Dev task
  * Runs a sequence of tasks on start
  */
 gulp.task('dev', gulp.series(
   cleanPublic,
   'compile',
-  'serve'
+  'watch',
```
Comment (domoscargin) [inline gulpfile.js:61]: Are there any "gotchas" we might miss running these tasks in slightly different order?

File: nodemon.json [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
   "exec": "node ./app/start.js",
   "watch": ["app", "src/govuk"],
+  "ignore": ["**/*.test.*"],
+  "ext": "js",
```
Comment (domoscargin) [inline nodemon.json:5]: Removing json I think will not catch the json file we have here: https://github.com/alphagov/govuk-frontend/blob/main/app/views/full-page-examples/search/data.json

But that's the only example I can find.

Comment (colinrotherham) [reply nodemon.json:5]: Ahh good detective work, thank you. Missed that one 👍

File: gulpfile.js (hunk 1 above)
Comment (colinrotherham) [reply gulpfile.js:61]: Not that I could see (as Nodemon does its own watching) and no compile steps are required for Express to start

Comment (colinrotherham) [reply gulpfile.js:61]: We still have `'compile'` coming first, so browser assets served by Express will already be in place too

File: nodemon.json (hunk 2 above)
Comment (colinrotherham) [reply nodemon.json]: Added `.json` again 😊

**Review by domoscargin — DISMISSED** [review]
LGTM!

**Review by domoscargin — COMMENTED** [review]
Sorry for pulling my approval last second!

Just want to check a workflow here.

It looks like now the `watch` task recompiles `.mjs` files to whichever folder is being used, but the app doesn't get restarted, since none of that should trigger `nodemon` right now.

So if a dev edits an `.mjs` file, currently they'd need to at least refresh the page to see the changes, right?

**Comment by colinrotherham** [comment]
@domoscargin Yeah that's it, totally separate now but same behaviour as before

Previously you would see erroneous Nodemon restarts for client-side JavaScript, but nothing server-side had actually changed

I did wonder if **livereload** or **Browser Sync** would be a handy improvement in future, but their JavaScript likely won't meet our browser support requirements

**Comment by domoscargin** [comment]
Ah great.

Had a quick play around and the previous set up still requires a refresh if you edit an `.mjs` file in `src`.

I had vague memories that the server restart process somehow got around that, and we were losing a bit of convenience, but that doesn't appear to be the case, so I'll approve again.

**Comment by colinrotherham** [comment]
This PR was related to https://github.com/alphagov/govuk-frontend/issues/2243

**Comment by colinrotherham** [comment]
Another one included by https://github.com/alphagov/govuk-frontend/issues/2243

**Pull request merged**